components/sst_importer/src/sst_importer.rs

```diff
@@ -451,28 +719,21 @@ impl SstImporter {
                 not_in_range += 1;
                 continue;
             }
-            let value = event_iter.value().to_vec();
-            build_fn(key.clone(), value);
 
-            let iter_key = key.clone();
-            smallest_key = smallest_key.map_or_else(
-                || Some(iter_key.clone()),
-                |v: Vec<u8>| Some(v.min(iter_key.clone())),
-            );
-
-            largest_key = largest_key.map_or_else(
-                || Some(iter_key.clone()),
-                |v: Vec<u8>| Some(v.max(iter_key.clone())),
-            );
+            build_fn(key.clone(), value);
+            smallest_key = smallest_key
+                .map_or_else(|| Some(key.clone()), |v: Vec<u8>| Some(v.min(key.clone())));
+            largest_key = largest_key
+                .map_or_else(|| Some(key.clone()), |v: Vec<u8>| Some(v.max(key.clone())));
         }
-        info!("build download request file done"; "total keys" => %total_key,
+        if total_key != not_in_range {
```

Comment:
when will this if condition be true?

Reply:
Some key-value events in files do not belong to this region[startkey, endkey).
This is just for a log.